source/controllers/src/impedance/CartesianTwistController.cpp

```diff
@@ -0,0 +1,120 @@
+#include "controllers/impedance/CartesianTwistController.hpp"
+
+namespace controllers::impedance {
+
+using namespace state_representation;
+
+CartesianTwistController::CartesianTwistController(double linear_principle_damping,
+                                                   double linear_orthogonal_damping,
+                                                   double angular_stiffness,
+                                                   double angular_damping) :
+    linear_principle_damping_(std::make_shared<Parameter<double>>("linear_principle_damping",
+                                                                  linear_principle_damping)),
+    linear_orthogonal_damping_(std::make_shared<Parameter<double>>("linear_orthogonal_damping",
+                                                                   linear_orthogonal_damping)),
+    angular_stiffness_(std::make_shared<Parameter<double>>("angular_stiffness", angular_stiffness)),
+    angular_damping_(std::make_shared<Parameter<double>>("angular_damping", angular_damping)),
+    dissipative_ctrl_(ComputationalSpaceType::LINEAR),
+    velocity_impedance_ctrl_(Eigen::MatrixXd::Zero(6, 6), Eigen::MatrixXd::Zero(6, 6)) {
+  set_angular_gains(angular_stiffness, angular_damping);
```

Comment:
I feel `set_linear_gains` is missing as it also set the damping matrix which is not covered by the constructor

Reply:
Indeed, great catch. I missed this because my unit test was passing, and indeed doing something like the following snippet still seemed valid:
```c++
CartesianTwistController controller(10, 20, 30, 40);
std::cout << controller.get_gains() << std::endl;
// >> 10 20 30 40
```

But, of course, this is an oversight. The `Parameter` for the linear gains was set by the constructor, so the `get_gains` function returns what you expect. Indeed though the dissipative controller does not have the correct gains.